_posts/2020-11-09-the-cloud-data-warehouse.md

```diff
@@ -0,0 +1,158 @@
+
+* working at the virtualization layer
+* low utilization of warm standbys
+* typically manual disaster recovery processes
+
+When you consider [availablity and recovery modes for BigQuery](https://cloud.google.com/bigquery/docs/availability) and [Snowflake's architecture](https://resources.snowflake.com/snowflake/how-to-make-data-protection-and-high-availability-for-analytics-fast-and-easy), it is clear that the discussion around disaster recovery changes, somewhat, for a number of reasons
+* the kind of disaster scenario you need to recover from may be different
+* the time and costs to rehearse disaster recovery are likely reduced
+* the skills required to recover from disaster may be different
+
+Does this mean disaster recovery goes away with a Cloud Data Warehouse? Not necessarily - but there's a good chance it will be different and easier to automate.
+
+### The Performance Equation
+
+A major advantage of Cloud Data Warehouses is their ability to compute queries that would typically fail on fixed size, in-memory query engines like Presto. That is not a criticism of Presto, but rather that at any one point in time, Presto is working with a fixed number of compute nodes. If you are using Presto on an AWS EMR cluster for example, you can [scale the cluster](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-automatic-scaling.html) in and out in response to scheduler memory availability - but there's no guarantee scaling will be effective before current queries run out of memory.
+
+Each implementation of a Cloud Data Warehouse is different, but using Snowflake as an example, they manage these kinds of challenges by [**spilling**](https://community.snowflake.com/s/article/Recognizing-Disk-Spilling) (a bit like swap file paging). When memory is exhausted, interim results are streamed to local disk, then if that becomes exhausted, it writes to remote storage (e.g. S3 / GCS) instead. With few exceptions, this will make your query a lot slower to execute, but at least you will get your results, and it can help you make decisions about whether to move your query to more powerful compute, as well as optimizing your query.
+
+> The **key question** to ask yourself is this: if you are concerned primarily with your **time to value**, where would you rather spend your effort - managing complex big data infrastructure? Or,  **generating value for your organisation from data**?
+
+
+Cloud Data Warehouses are designed to manage high volume *concurrent workloads*. Most organisations do not have steady workloads that balance their compute requirements over the day or week - it is typical for workloads to be *spiky*, instead. If you have a workload that is subject to regular spikes, or encounters *mega-events* (such as sale days, race events or EOFY reporting), how do you ensure quality of service that users expect?
```

Comment:
I think this is the point of this section. 

I would move this whole section up to para 1, and move the first two paragraphs down under "Both platforms..."

Reply:
i could do that. feels then that the title "The Performance Equation" does not leading naturally into the flow, however